web/Gruntfile.js

```diff
@@ -40,8 +45,12 @@ module.exports = function (grunt) {
                     '<%= config.srcScriptsDir %>/**/*.es.js'
                 ],
                 tasks: [
-                    'compilejs'
+                    'build'
                 ]
+            },
+            ngtemplates: {
+                files: ['app/templates/**/*.html'],
+                tasks: ['build']
```

Comment:
Sure; I think there should be a separate watcher, so html changes runs `ngtemplates`, then you have a watcher on all the source files to run `build`.